keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -452,94 +457,194 @@ change are understandable. This may include API specs (though not always
+
 #### Container Lifecycle 
 - The container's lifecycle will be managed by the container runtime just like other Windows container types.
-#### Container users 
--  The privileged container should be able to run as any user that's available on the host or in the domain of the host machine. Password accounts are being investigated.
+
+#### Container users
+
+- The privileged container can run as any user that's available on the host or in the domain of the host machine. Password accounts are being investigated.
+Running privileged containers as non SYSTEM/admin accounts will be the primary way operators can restrict access to system resources (files, registry, named pipes, WMI, etc).
+More information on Windows resource access can be found at https://docs.microsoft.com/en-us/archive/msdn-magazine/2008/november/access-control-understanding-windows-file-and-registry-permissions.
+
 #### Container Mounts
-- Directory mounts (i.e. directory C:\test mapped to C:\test in the privileged container) are still being investigated. Note that file system mapping in the current implementation would expose both the host filesystem and the container filesystem to the processes in the privileged container. We are still investigation how external mounts such as secrets or storage accounts would be added. The default visibility of all the host filesystem may present backwards compatibility challenges in the future if host path mounting is enabled in Windows. Currently the only method to restrict would be via the user in a way similar to [UID or GID](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) in the PSP. Currently however, this host path mounting is unlikely to be enabled in the future as a restriction of Job Objects.  
+
+- When privileged containers are started a new volume will be created on the host which will contain the contains of the container image. Containers will have a default working directory that points to this container volume. Containers will also have full access to the host file-system (unless restricted by filed-based ACLs and the run_as_username used to start the container.) Processes should use absolute paths when accessing files on the host and relative paths when accessing files brought in via the container image.
+Note: there will be no `chroot` equivalent
+- Volume mounts (including service account tokens) are expected to work for privileged containers but may not be supported in alpha. The current plan is to mount these volumes under the volume created for the container. This will be re-visited for beta and extensively documented.
```

Comment:
It would be good to get review of the mount implications by sig-storage, especially if the expected changes will involve significant changes to `k8s.io/mount-utils` or kubelet volume code.

> Volume mounts (including service account tokens) are expected to work for privileged containers but may not be supported in alpha.

Will lack of service account token support make it hard to get feedback on the feature? Several of the daemonset examples cited in the motivation/goals sections require service account credentials. That seems like a significant gap to me that should be solved while in alpha to get feedback on the complete feature before moving forward.

> Processes should use absolute paths when accessing files on the host and relative paths when accessing files brought in via the container image.

The client libraries I'm aware of look for auto-mounted credentials at fixed paths (e.g. https://pkg.go.dev/k8s.io/client-go/rest#InClusterConfig). Does that work with regular Windows containers today? If so, what is the plan for allowing those libraries to work in a hostProcess environment?

Reply:
I updated this section.